### ISSUE TITLE
SpecialFunctions compat

### ISSUE THREAD

**Issue by mloubout** [tknopp/NFFT.jl#52]
Can the compat for SpecialFunctions be updated?

CUDA.jl version >= is only compatible with SpecialFunctions.jl=1.3 so the current restriction to 1.0 is preventing working with more recent CUDA versions. (2.6.3 at most currently).

**Comment by tknopp** [comment]
I don't get the problem. Combat 1.0 means 1.0 - 2.0 or not? See https://pkgdocs.julialang.org/v1/compatibility/

Here, I have
```
[052768ef] CUDA v2.6.3
[efe261a4] NFFT v0.6.1
[276daf66] SpecialFunctions v1.6.0
```
Not sure what holds back CUDA though.

**Comment by mloubout** [comment]
So to double check I started a fresh julia with nothing but NFFT

```
(@v1.6) pkg> status
      Status `~/.julia/environments/v1.6/Project.toml`
  [efe261a4] NFFT v0.6.1

(@v1.6) pkg> add CUDA@3
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package SpecialFunctions [276daf66]:
 SpecialFunctions [276daf66] log:
 ├─possible versions are: 0.7.0-1.6.1 or uninstalled
 ├─restricted by compatibility requirements with CUDA [052768ef] to versions: 1.3.0-1.6.1
 │ └─CUDA [052768ef] log:
 │   ├─possible versions are: 0.1.0-3.4.1 or uninstalled
 │   └─restricted to versions 3 by an explicit requirement, leaving only versions 3.0.0-3.4.1
 └─restricted by compatibility requirements with NFFT [efe261a4] to versions: [0.7.0-0.8.0, 0.10.0-0.10.3] — no versions left
   └─NFFT [efe261a4] log:
     ├─possible versions are: 0.3.0-0.6.1 or uninstalled
     ├─restricted to versions * by an explicit requirement, leaving only versions 0.3.0-0.6.1
     └─restricted by compatibility requirements with CUDA [052768ef] to versions: 0.3.0-0.5.2 or uninstalled, leaving only versions: 0.3.0-0.5.2
       └─CUDA [052768ef] log: see above
```

And it does fail to work with CUDA@3 because of SpecialFunctions. 

Also it looks like the `Project.toml` update with CUDA 3.1 hasn't been released yet.

**Comment by tknopp** [comment]
Ah, ok then everything is clear. I still merged the PR and will do a release next.

**Comment by mloubout** [comment]
Fixed in #53 closing. Thanks.
